### ISSUE TITLE
Linker emits wrong `standalone: true` for pre-v19 NgModule-declared components/directives/pipes

### ISSUE THREAD

**Issue by tomer953** [voidzero-dev/oxc-angular-compiler#87]
## Summary

The OXC linker unconditionally defaults `standalone` to `true` when the `isStandalone` property is absent from a partial declaration. However, Angular v12–v18 libraries default `standalone` to `false`. Only v19+ defaults to `true`.

This causes all NgModule-declared components from pre-v19 libraries (e.g., PrimeNG v12, Angular Material v15) to be incorrectly marked as `standalone: true` at runtime.

## Runtime Error

```
NG0303: Can't bind to 'ngIf' since it isn't a known property of 'div'
(used in the 'OverlayPanel' component template).
```

This occurs because:
- The component is incorrectly marked `standalone: true`, so Angular treats it as standalone
- Standalone components must declare their own dependencies — but the NgModule-declared component relies on its NgModule's compilation scope
- Angular has no way to resolve `NgIf`, `NgClass`, etc.

## Root Cause

In [`crates/oxc_angular_compiler/src/linker/mod.rs`](crates/oxc_angular_compiler/src/linker/mod.rs), three functions unconditionally default `standalone` to `true`:

**`link_component`:**
```rust
let standalone = get_bool_property(meta, "isStandalone").unwrap_or(true);  // ❌ Always true
```

**`link_directive`:**
```rust
let standalone = get_bool_property(meta, "isStandalone").unwrap_or(true);  // ❌ Always true
```

**`link_pipe`:**
```rust
let standalone = get_property_source(meta, "isStandalone", source).unwrap_or("true");  // ❌ Always true
```

## Expected Behavior (Angular Reference Linker)

The Angular reference linker in [`packages/compiler-cli/linker/src/file_linker/partial_linkers/util.ts`](crates/oxc_angular_compiler/angular/packages/compiler-cli/linker/src/file_linker/partial_linkers/util.ts:122) uses version-aware defaulting:

```typescript
const STANDALONE_IS_DEFAULT_RANGE = new semver.Range(`>= 19.0.0 || 0.0.0-PLACEHOLDER`);

export function getDefaultStandaloneValue(version: string): boolean {
  return STANDALONE_IS_DEFAULT_RANGE.test(version);
}
```

- `version >= 19.0.0` → default `standalone: true`
- `version < 19.0.0` → default `standalone: false`
- `version == "0.0.0-PLACEHOLDER"` → default `standalone: true` (latest/head)

### Input (PrimeNG v12 partial declaration):
```javascript
OverlayPanel.ɵcmp = i0.ɵɵngDeclareComponent({
    minVersion: "12.0.0",
    version: "12.0.5",        // ← v12, no isStandalone field
    type: OverlayPanel,
    selector: "p-overlayPanel",
    template: `<div *ngIf="render">...</div>`,
    directives: [{ type: i1.NgIf, selector: "[ngIf]", ... }]
});
```

### Current output (wrong ❌):
```javascript
OverlayPanel.ɵcmp = ɵɵdefineComponent({
    standalone: true,   // ❌ Should be false for v12
    ...
});
```

### Expected output (correct ✅):
```javascript
OverlayPanel.ɵcmp = ɵɵdefineComponent({
    standalone: false,  // ✅ v12 defaults to false
    ...
});
```

## Fix

Add a `get_default_standalone_value()` helper that reads the `version` property from the partial declaration metadata:

```rust
fn get_default_standalone_value(meta: &ObjectExpression<'_>) -> bool {
    if let Some(version_str) = get_string_property(meta, "version") {
        if version_str == "0.0.0-PLACEHOLDER" {
            return true;
        }
        let base_version = version_str.split('-').next().unwrap_or(version_str);
        if let Some(major_str) = base_version.split('.').next() {
            if let Ok(major) = major_str.parse::<u32>() {
                return major >= 19;
            }
        }
    }
    true // If we can't determine the version, default to true (latest behavior)
}
```

Update all three call sites:
```rust
let standalone = get_bool_property(meta, "isStandalone")
    .unwrap_or_else(|| get_default_standalone_value(meta));
```

## Suggested Tests

```rust
#[test]
fn test_link_component_v12_defaults_standalone_false() {
    let allocator = Allocator::default();
    let code = r#"
import * as i0 from "@angular/core";
export class MyComponent {}
MyComponent.ɵcmp = i0.ɵɵngDeclareComponent({
    minVersion: "12.0.0", version: "12.0.5", ngImport: i0,
    type: MyComponent, selector: "my-comp", template: "<div>Hello</div>"
});
"#;
    let result = link(&allocator, code, "test.mjs");
    assert!(result.linked);
    assert!(
        result.code.contains("standalone: false"),
        "v12 component without isStandalone should default to false"
    );
}

#[test]
fn test_link_component_v18_defaults_standalone_false() {
    let allocator = Allocator::default();
    let code = r#"
import * as i0 from "@angular/core";
export class MyComponent {}
MyComponent.ɵcmp = i0.ɵɵngDeclareComponent({
    minVersion: "14.0.0", version: "18.2.0", ngImport: i0,
    type: MyComponent, selector: "my-comp", template: "<div>Hello</div>"
});
"#;
    let result = link(&allocator, code, "test.mjs");
    assert!(result.linked);
    assert!(
        result.code.contains("standalone: false"),
        "v18 component without isStandalone should default to false"
    );
}

#[test]
fn test_link_component_v19_defaults_standalone_true() {
    let allocator = Allocator::default();
    let code = r#"
import * as i0 from "@angular/core";
export class MyComponent {}
MyComponent.ɵcmp = i0.ɵɵngDeclareComponent({
    minVersion: "14.0.0", version: "19.0.0", ngImport: i0,
    type: MyComponent, selector: "my-comp", template: "<div>Hello</div>"
});
"#;
    let result = link(&allocator, code, "test.mjs");
    assert!(result.linked);
    assert!(
        result.code.contains("standalone: true"),
        "v19 component without isStandalone should default to true"
    );
}

#[test]
fn test_link_directive_v12_defaults_standalone_false() {
    let allocator = Allocator::default();
    let code = r#"
import * as i0 from "@angular/core";
export class NgIf {}
NgIf.ɵdir = i0.ɵɵngDeclareDirective({
    minVersion: "12.0.0", version: "12.0.5", ngImport: i0,
    type: NgIf, selector: "[ngIf]"
});
"#;
    let result = link(&allocator, code, "common.mjs");
    assert!(result.linked);
    assert!(
        result.code.contains("standalone: false"),
        "v12 directive without isStandalone should default to false"
    );
}

#[test]
fn test_link_pipe_v12_defaults_standalone_false() {
    let allocator = Allocator::default();
    let code = r#"
import * as i0 from "@angular/core";
export class AsyncPipe {}
AsyncPipe.ɵpipe = i0.ɵɵngDeclarePipe({
    minVersion: "12.0.0", version: "12.0.5", ngImport: i0,
    type: AsyncPipe, name: "async"
});
"#;
    let result = link(&allocator, code, "common.mjs");
    assert!(result.linked);
    assert!(
        result.code.contains("standalone: false"),
        "v12 pipe without isStandalone should default to false"
    );
}
```

## Impact

Affects **every Angular library** compiled with v12–v18 that uses NgModule-based architecture, including PrimeNG, Angular Material (older versions), and any third-party library with NgModule declarations.
